contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/wifi_test.py

```diff
@@ -0,0 +1,296 @@
+#!/usr/bin/env python3
+import argparse
+import subprocess
+import sys
+import time
+import logging
+import re
+
+# Configure logging
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
+)
+
+
+class WiFiManager:
+    def __init__(
+        self,
+        interface,
+        type,
+        mode,
+        band,
+        channel,
+        key_mgmt,
+        group,
+        peer,
+        ssid,
+        ssid_pwd,
+    ):
+        self._command = "nmcli"
+        self.interface = interface
+        self.type = type
+        self.mode = mode
+        self.band = band
+        self.channel = channel
+        self.key_mgmt = key_mgmt
+        self.group = group
+        self.ssid = ssid if ssid is not None else "qa-test-ssid"
+        self.ssid_pwd = ssid_pwd if ssid_pwd is not None else "insecure"
+        self.peer = peer
+        self.conname = "qa-test-ap"
+
+    def run_command(self, command):
+        try:
+            output = subprocess.check_output(
+                command, shell=True, stderr=subprocess.STDOUT
+            )
+            return output.decode()
+        except subprocess.CalledProcessError as e:
+            logging.error("Command failed: %s", e.output.decode())
+            raise
```

Comment:
I would suggest we move the run_command out of the `WiFiManager` class, the reason is 
1. the WiFi manager is uses the run_command function to configure wireless settings only.
2. we could re-use it in the `ping_test` and `connect_host_device`

Reply:
I've pushed the fix